zingolib/src/wallet/notes.rs

```diff
@@ -0,0 +1,491 @@
+    type Diversifier = zcash_primitives::sapling::Diversifier;
+    type Note = zcash_primitives::sapling::Note;
+    type Node = zcash_primitives::sapling::Node;
+    type Nullifier = zcash_primitives::sapling::Nullifier;
+
+    fn diversifier(&self) -> &Self::Diversifier {
+        &self.diversifier
+    }
+
+    fn nullifier_mut(&mut self) -> &mut Option<Self::Nullifier> {
+        &mut self.nullifier
+    }
+
+    fn from_parts(
+        diversifier: zcash_primitives::sapling::Diversifier,
+        note: zcash_primitives::sapling::Note,
+        witnessed_position: Option<Position>,
+        nullifier: Option<zcash_primitives::sapling::Nullifier>,
+        spent: Option<(TxId, u32)>,
+        unconfirmed_spent: Option<(TxId, u32)>,
+        memo: Option<Memo>,
+        is_change: bool,
+        have_spending_key: bool,
+        output_index: Option<u32>,
+    ) -> Self {
+        Self {
+            diversifier,
+            note,
+            witnessed_position,
+            nullifier,
+            spent,
+            unconfirmed_spent,
+            memo,
+            is_change,
+            have_spending_key,
+            output_index,
+        }
+    }
+
+    fn get_deprecated_serialized_view_key_buffer() -> Vec<u8> {
+        vec![0u8; 169]
+    }
+
+    fn have_spending_key(&self) -> bool {
+        self.have_spending_key
+    }
+
+    fn is_change(&self) -> bool {
+        self.is_change
+    }
+
+    fn is_change_mut(&mut self) -> &mut bool {
+        &mut self.is_change
+    }
+
+    fn memo(&self) -> &Option<Memo> {
+        &self.memo
+    }
+
+    fn memo_mut(&mut self) -> &mut Option<Memo> {
+        &mut self.memo
+    }
+
+    fn note(&self) -> &Self::Note {
+        &self.note
+    }
+
+    fn nullifier(&self) -> Option<Self::Nullifier> {
+        self.nullifier
+    }
+
+    fn pool() -> Pool {
+        Pool::Sapling
+    }
+
+    fn spent(&self) -> &Option<(TxId, u32)> {
+        &self.spent
+    }
+
+    fn spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
+        &mut self.spent
+    }
+
+    fn transaction_metadata_notes(wallet_transaction: &TransactionRecord) -> &Vec<Self> {
+        &wallet_transaction.sapling_notes
+    }
+
+    fn transaction_metadata_notes_mut(
+        wallet_transaction: &mut TransactionRecord,
+    ) -> &mut Vec<Self> {
+        &mut wallet_transaction.sapling_notes
+    }
+
+    fn pending_spent(&self) -> &Option<(TxId, u32)> {
+        &self.unconfirmed_spent
+    }
+
+    fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
+        &mut self.unconfirmed_spent
+    }
+
+    fn value_from_note(note: &Self::Note) -> u64 {
+        note.value().inner()
+    }
+
+    fn witnessed_position(&self) -> &Option<Position> {
+        &self.witnessed_position
+    }
+
+    fn witnessed_position_mut(&mut self) -> &mut Option<Position> {
+        &mut self.witnessed_position
+    }
+
+    fn output_index(&self) -> &Option<u32> {
+        &self.output_index
+    }
+}
+
```

Comment:
outputindex_mut isnt used anywhere? nice find!

Reply:
credit to @AloeareV